icu4c/source/common/caniter.cpp

```diff
@@ -490,6 +493,13 @@ Hashtable *CanonicalIterator::getEquivalents2(Hashtable *fillinResult, const cha
 
                 ne = remainder.nextElement(el);
             }
+            // ICU-22642 Guards against strings that have too many permutations
```

Comment:
Either:

> Guards against strings that have so many permutations that they would otherwise hang the function.


Or:

> Guards against strings that have too many permutations and would otherwise hang the function.